include/swift/AST/DiagnosticsDriver.def

```diff
@@ -183,23 +183,28 @@ WARNING(warn_use_filelists_deprecated, none,
         "'-driver-filelist-threshold=0' instead", ())
 
 WARNING(warn_unable_to_load_swift_ranges, none,
-"unable to load swift ranges file \"%0\", %1",
-(StringRef, StringRef))
+       "unable to load swift ranges file \"%0\", %1",
+       (StringRef, StringRef))
 
 WARNING(warn_unable_to_load_compiled_swift, none,
-"unable to load previously compiled swift file \"%0\", %1",
-(StringRef, StringRef))
+       "unable to load previously compiled swift file \"%0\", %1",
+       (StringRef, StringRef))
 
 WARNING(warn_unable_to_load_primary, none,
-"unable to load primary swift file \"%0\", %1",
-(StringRef, StringRef))
+       "unable to load primary swift file \"%0\", %1",
+       (StringRef, StringRef))
 
 ERROR(cannot_find_migration_script, none,
       "missing migration script from path '%0'", (StringRef))
 
 ERROR(error_darwin_static_stdlib_not_supported, none,
       "-static-stdlib is no longer supported on Apple platforms", ())
 
+WARNING(warn_library_evolution_test_mode_compatibility, none,
```

Comment:
I think we need a stronger wording around this, and to possibly make it a hard error (but that’s a different discussion).

> specifying -enable-library-evolution and -enable-testing together will cause program instability

We can attach notes with more info depending on if we know we’re targeting a release build or not. e.g. release build suggests the client wants -enable-library-evolution, and debug suggests they want -enable-testing.

Reply:
Thanks @CodaFi :))
> specifying -enable-library-evolution and -enable-testing together will cause program instability
Sure, this one seems better good :) 

> We can attach notes with more info depending on if we know we’re targeting a release build or not. e.g. release build suggests the client wants -enable-library-evolution, and debug suggests they want -enable-testing.

Humm... maybe a 'consider removing -enable-testing' if is a release build and the other for debug?

Reply:
Also, how do we know it is a release/debug compiler invocation? `swiftc` has a flag or arg for that?

Reply:
Also, how do we know it is a release/debug compiler invocation? swiftc has a flag or arg for that?

Reply:
Huh, I actually think the original message was closer to the truth. The specific problems we're trying to warn of are:

* `@testable import` won't import any internal APIs if the module is loaded from a module interface (swiftinterface).
* `@testable import` may cause compiler crashes if the module is loaded from a compiled module interface (swiftmodule).

Here's the wording I would suggest (the `StringRef` is the module name):

```
WARNING(warn_library_evolution_test_mode_compatibility, none,
        "testability is not compatible with library evolution and may cause "
        "build failures in clients using '@testable import %0'; consider "
        "disabling testability and testing only public APIs or disabling "
        "library evolution in builds used for testing", (StringRef))
```

(However, I know that @slavapestov is skeptical of this whole idea. Perhaps, early in the new year, we can hash all this out.)